### PR TITLE
webserver: reorganize and refactor javascript

### PR DESCRIPTION
**Pull request by buck54321**
Creates a series of page classes that inherit from a `BasePage` class. Each page is moved to its own module. Common functions and constants are also organized into new or existing modules. 

This PR does not move or fully refactor the handling for the markets page. I'd like to follow up with that in a separate PR that focuses on /markets.

No substantive changes, but a couple of bug fixes and a bit more documentation.

### CODE REVIEW

**Comment by buck54321** [comment]
Nope. That's a bug. Fix in #256. Also, you appear to have nothing in that wallet/account. Is that expected?

File: client/webserver/site/src/js/login.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (pw === '') {
+      page.errMsg.textContent = 'password cannot be empty'
+      Doc.show(page.errMsg)
+      return
+    }
+    app.loaded()
+    var res = await postJSON('/api/login', { pass: pw })
+    if (!app.checkResponse(res)) return
+    res.notes.reverse()
+    app.setNotes(res.notes)
```
Comment (chappjc) [inline client/webserver/site/src/js/login.js:35]: At first I was confused about `res.notes`, but I see that it's the queued notifications from the server, right?

File: client/webserver/site/src/js/login.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    await app.fetchUser()
+    app.setLogged(true)
+    app.loadPage('markets')
```
Comment (chappjc) [inline client/webserver/site/src/js/login.js:38]: I feel like this should depend on the state of the wallets, but it's reasonable.

File: client/webserver/site/src/js/app.js [hunk 3]
```diff
@@ -283,233 +336,47 @@ export default class Application {
...
+
+  /*
+   * checkResponse checks the response object as returned from the functions in
+   * the http module. If the response indicates that the request failed, a
+   * message will be displayed in the drop-down notifications and false will be
+   * returned.
+   */
+  checkResponse (resp) {
+    if (!resp.requestSuccessful || !resp.ok) {
+      if (this.user.inited) this.notify(ntfn.make('API error', resp.msg, ntfn.ERROR))
```
[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/app.js:348]: Surprised you caught that. The entire browser notification cache is replace on sign in, but not on registration, and the notifications are shown after both. So if you don't clear the cache, wipe your DB, and reregister, you will see notifications from before until you actually sign out and sign back in again.

File: client/webserver/site/src/js/app.js [hunk 4]
```diff
@@ -28,30 +28,46 @@ const notificationRoute = 'notify'
...
     await this.fetchUser()
+    // Handle back navigation from the browser.
     bind(window, 'popstate', (e) => {
       const page = e.state.page
       if (!page && page !== '') return
       this.loadPage(page)
     })
+    // The main element is the interchangable part of the page that doesn't
+    // include the header. Main should define a data-handler attribute
+    // associated with  one of the available constructors.
```
[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/app.js:56]: Hmm. I think that might be an artifact.

File: client/webserver/site/src/js/login.js (hunk 1 above)
Comment (buck54321) [reply client/webserver/site/src/js/login.js:35]: Yep. The entire notification cache is replaced on login.

File: client/webserver/site/src/js/http.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+/*
+ * requestJSON encodes the object and sends the JSON to the specified address.
+ */
+export async function requestJSON (method, addr, reqBody) {
+  try {
+    const response = await window.fetch(addr, {
+      method: method,
+      headers: new window.Headers({ 'content-type': 'application/json' }),
```
[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/http.js:8]: These are request headers, so it's really up to the server what to do with the info, and we control the server. It's good practice though, so I can add it.

[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/http.js:8]: Actually, nevermind. [This RFC](https://tools.ietf.org/html/rfc4627#section-3) defines the default for `application/json` as utf-8, and (This IANA registration)[https://www.iana.org/assignments/media-types/application/json] says 

> Note:  No "charset" parameter is defined for this registration. Adding one really has no effect on compliant recipients.

File: client/webserver/site/src/js/app.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/app.js:348]: I'm gonna move this to `RegisterPage` though.

**Pull request merged**